automation/validate-pvc-name-stability.py

```diff
@@ -0,0 +1,105 @@
+#! python
+
+import logging
+import os
+import os.path
+import yaml
+import subprocess
+import sys
+
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+
+
+def validatePVCNames(path, liveTemplates):
+    templates = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]
+    errors = []
+    for templateFilename in templates:
+        with open(os.path.join(path, templateFilename), 'r') as stream:
+            try:
+                template = yaml.safe_load(stream)
+
+                if template == None:
+                    logging.info("Empty template file: %s", template)
+                    continue
+
+                templateName = template["metadata"]["name"]
+                logging.info("Checking PVC name stability for: {}".format(templateName))
+
+                pvcName = getPVCNameFrom(template)
+                pvcNamespace = getPVCNamespaceFrom(template)
+
+                matchingLiveTemplate = liveTemplates.get(templateName, None)
+                if matchingLiveTemplate:
+                    liveTemplatePVCName = getPVCNameFrom(matchingLiveTemplate)
+                    liveTemplatePVCNamespace = getPVCNamespaceFrom(matchingLiveTemplate)
+
+                    if pvcName != liveTemplatePVCName:
+                        errors.append(pvcNameErrorMessage(pvcName, templateName))
+                    if pvcNamespace != liveTemplatePVCNamespace:
+                        errors.append(pvcNamespaceErrorMessage(pvcNamespace, templateName))
+                else:
+                    logging.info("Missing liveTemplate for {}".format(templateName))
+
+            except yaml.YAMLError as exc:
+                raise exc
+            except Exception as exc:
+                logging.error(exc)
+
+    if len(errors) > 0:
+        error_message = "\n".join(errors)
+        logging.info(FAILED_INFO_MESSAGE)
+        raise Exception(error_message)
+
+
+def getParamFrom(template, paramName):
+    for param in template["parameters"]:
+        if param["name"] == paramName:
+            return param["value"]
```

Comment:
If this function does not find the parameter `paramName`, it implicitly returns `None`.
It it is expected, please add an explicit `return None` at the end, to make it obvious.